server/src/main/java/org/opensearch/rest/RestController.java

```diff
@@ -505,6 +516,42 @@ private void handleBadRequest(String uri, RestRequest.Method method, RestChannel
         }
     }
 
+    /**
+     * Attempts to extracts auth token and login.
+     * 
+     * @returns false if there was an error and the request should not continue being dispatched
+     * */
+    private boolean handleAuthenticateUser(final RestRequest request, final RestChannel channel) {    
+        if (!FeatureFlags.isEnabled(FeatureFlags.IDENTITY)) {
+            return true;
+        }
```

Comment:
I would prefer moving this to L406, lets not get to `handleAuthenticateUser` if the feature is not enabled.